tests/toranj/start.sh

```diff
@@ -58,15 +58,13 @@ run() {
             return
         fi
 
-        # On Travis, we allow a failed test to be retried up to 3 attempts.
-        if [ "$BUILD_TARGET" = "toranj-test-framework" ]; then
-            if [ "$counter" -lt 2 ]; then
```

Comment:
Can we make the `toranj` tests not fail in the first place?

Reply:
@simonlingoogle, that's always the main goal and and  in many cases when investigating why some test fail rarely, we find some issue/bug in the some part of the the code that we fix (in some cases it's an issue in core OT, or sometime in the platform layer simulation or driver/wpantund). 
In some cases it can be assumption in the way the test script is written which we address (a good example of this case is #4516).

That said, the retries are for issues/failures that are usually beyond our immediate control and  may be related to OS/processes/sockets/dbus/pipes, etc. For example, socket tx (which we use for posix simulation) is not guaranteed to be 100% reliable, or IP messages can be dropped/delayed by netif (used by wpantund), or port number being blocked (test cannot bind to randomly selected port number). Many of these are rare but possible (e.g., perfectly valid for linux `netif` to drop an IPv6 message). 

Note that some of the tests run with very high speed-up generating a lot of events/actions that can overflow system/processes (e.g., the `test-602-channel-manager-channel-select.py` emulates about 12 hours of network operation within 4.5s, the test can be made more robust by running it at slower speed up  but at the end, it made more sense to run it fast (to help make the average test run time shorter) and allow retry for occasional failures.